File: C/beginner/1079.c
// https://judge.beecrowd.com/pt/problems/view/1079
// Médias Ponderadas

// Libraries
#include <stdio.h>

// Main Function
int main(){
    // Starting variables
    int usr_sequences;
    double usr_v1, usr_v2, usr_v3;

    // Asking user for sequences
    scanf("%d", &usr_sequences);

    // Iterating between sequences
    for (int i = 0; i < usr_sequences; i++){
        // Requesting values
        scanf("%lf %lf %lf", &usr_v1, &usr_v2, &usr_v3);
        // Printing result
        printf("%.1lf\n", (usr_v1 * 2 + usr_v2 * 3 + usr_v3 * 5) / 10);

    }
    
    // Ending Code
    return 0;
}

// By Fernando Serra
// https://github.com/fernandosserra

File: Haskell/beginner/1079.hs
-- https://judge.beecrowd.com/pt/problems/view/1079
-- Médias Ponderadas

-- Main function
main :: IO ()
main = do
    -- Reading the number of sequences
    usr_sequences <- readLn
    -- Iterating through the sequences
    sequence_ (replicate usr_sequences processSequence)

-- Function to process each sequence
processSequence :: IO ()
processSequence = do
    -- Reading the values
    [usr_v1, usr_v2, usr_v3] <- fmap (map read . words) getLine
    -- Calculating and printing the result
    let result = (usr_v1 * 2 + usr_v2 * 3 + usr_v3 * 5) / 10
    print (roundTo1Decimal result)

-- Function to round to one decimal place
roundTo1Decimal :: Double -> Double
roundTo1Decimal x = fromIntegral (round (x * 10)) / 10

-- By Fernando Serra
-- https://github.com/fernandosserra